game/factions/faction.py

```diff
@@ -137,12 +118,64 @@ class Faction:
     #: both will use it.
     unrestricted_satnav: bool = False
 
-    def has_access_to_unittype(self, unit_class: GroundUnitClass) -> bool:
-        for vehicle in itertools.chain(self.frontline_units, self.artillery_units):
-            if vehicle.unit_class is unit_class:
-                return True
+    # Save all accessible units for performance increase
+    _accessible_units: list[UnitType[Any]] = field(default_factory=list)
+
+    def __getitem__(self, item: str) -> Any:
+        return getattr(self, item)
+
+    @property
+    def accessible_units(self) -> Iterator[UnitType[Any]]:
+        if not self._accessible_units:
+            self._load_accessible_units()
+        yield from self._accessible_units
+
+    @property
+    def air_defenses(self) -> list[str]:
+        """Returns the Air Defense types"""
+        air_defenses = [a.name for a in self.air_defense_units]
+        air_defenses.extend(
+            [pg.name for pg in self.preset_groups if pg.role == GroupRole.AntiAir]
+        )
+        return sorted(air_defenses)
+
+    def has_access_to_unit_type(self, unit_type: str) -> bool:
```

Comment:
i think the misleading part here is the unit_type naming in the function. what it actually means is the dcs type id. i will change this by reorganizing the template.py to use the actual UnitType. Again same answer: this was meant to be used without already migrated units. As all are migrated already it can be changed from str to UnitType :)